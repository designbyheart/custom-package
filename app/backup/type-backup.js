import type { ReactNavigation } from '../common/type-common'
import type { CustomError } from '../common/type-common'
import type { IsValid } from '../components/input-control/type-input-control'

export type GenerateRecoveryPhraseProps = {
  recoveryPassphrase: string,
} & ReactNavigation

export type GenerateRecoveryPhraseState = {}

export type VerifyRecoveryPhraseProps = {} & ReactNavigation

export type VerifyRecoveryPhraseState = {
  recoveryPassphrase: string,
  error: boolean,
}

export type ExportBackupFileProps = {
  walletBackup: () => void,
  backupPath: string,
} & ReactNavigation

export type ExportBackupFileState = {
  submitButtonText: string,
}

export type BackupCompleteProps = {} & ReactNavigation

export type BackupCompleteState = {
  submitButtonText: string,
  recoveryPassphrase: string,
}

export const BACKUP_STORE_STATUS = {
  IDLE: 'IDLE',
  GENERATE_PHRASE: 'GENERATE_PHRASE',
  GENERATE_BACKUP_FILE: 'GENERATE_BACKUP_FILE',
  EXPORT_BACKUP: 'EXPORT_BACKUP',
  BACKUP_COMPLETE: 'BACKUP_COMPLETE',
}

export type Passphrase = {
  phrase: string,
  salt: string,
  hash: string,
}

export type BackupStore = {
  passphrase: Passphrase,
  backupWalletPath: string,
  showBanner: boolean,
  lastSuccessfulBackup: string,
  isLoading: boolean,
} & StoreError &
  BackupStoreStatus

export type StoreError = { error: ?CustomError }
export type BackupStoreStatus = { status: $Keys<typeof BACKUP_STORE_STATUS> }

export const ERROR_BACKUP_WALLET = {
  code: 'WB-001',
  message: 'Error while backing up wallet',
}

export const ERROR_BACKUP_WALLET_SHARE = {
  code: 'WB-002',
  message: 'Error while sharing zipped backup',
}

export const ERROR_HYDRATING_BACKUP = {
  code: 'WB-004',
  message: 'Error hydrating backup',
}

export const START_BACKUP = 'START_BACKUP'
export const GENERATE_BACKUP_FILE = 'GENERATE_BACKUP_FILE'
export const GENERATE_BACKUP_FILE_SUCCESS = 'GENERATE_BACKUP_FILE_SUCCESS'
export const BACKUP_WALLET_FAIL = 'BACKUP_WALLET_FAIL'
export const GENERATE_RECOVERY_PHRASE = 'GENERATE_RECOVERY_PHRASE'
export const GENERATE_RECOVERY_PHRASE_SUCCESS =
  'GENERATE_RECOVERY_PHRASE_SUCCESS'
export const EXPORT_BACKUP = 'EXPORT_BACKUP'
export const EXPORT_BACKUP_SUCCESS = 'EXPORT_BACKUP_SUCCESS'
export const BACKUP_COMPLETE = 'BACKUP_COMPLETE'
export const HYDRATE_BACKUP = 'HYDRATE_BACKUP'
export const HYDRATE_BACKUP_FAILURE = 'HYDRATE_BACKUP_FAILURE'
export const PROMPT_WALLET_BACKUP_BANNER = 'PROMPT_WALLET_BACKUP_BANNER'

export type BackupStartAction = {
  type: typeof START_BACKUP,
  status: BACKUP_STORE_STATUS,
  isLoading: boolean,
  error: CustomError,
}

export type GenerateBackupFileAction = {
  type: typeof GENERATE_BACKUP_FILE_LOADING,
  status: BACKUP_STORE_STATUS,
  isLoading: boolean,
  error: CustomError,
}

export type GenerateBackupFileSuccessAction = {
  type: typeof GENERATE_BACKUP_FILE_SUCCESS,
  backupWalletPath: string,
  isLoading: boolean,
  error: CustomError,
}

export type BackupWalletFailAction = {
  type: typeof BACKUP_WALLET_FAIL,
  status: BACKUP_STORE_STATUS,
  isLoading: boolean,
  error: CustomError,
}

export type GenerateRecoveryPhraseAction = {
  type: typeof GENERATE_RECOVERY_PHRASE,
  status: BACKUP_STORE_STATUS,
  isLoading: boolean,
  error: CustomError,
}

export type GenerateRecoveryPhraseSuccessAction = {
  type: typeof GENERATE_RECOVERY_PHRASE_SUCCESS,
  status: BACKUP_STORE_STATUS,
  isLoading: boolean,
  passphrase: Passphrase,
  error: CustomError,
}

export type ExportBackupAction = {
  type: typeof EXPORT_BACKUP,
  status: BACKUP_STORE_STATUS,
  backupWalletPath: string,
  isLoading: boolean,
  error: CustomError,
}

export type ExportBackupSuccessAction = {
  type: typeof EXPORT_BACKUP_SUCCESS,
  status: BACKUP_STORE_STATUS,
  isLoading: boolean,
  error: CustomError,
}

export type BackupCompleteAction = {
  type: typeof BACKUP_COMPLETE,
  status: BACKUP_STORE_STATUS,
}

export type HydrateBackupAction = {
  type: typeof HYDRATE_BACKUP,
  status: BACKUP_STORE_STATUS,
}

export type HydrateBackupFailAction = {
  type: typeof HYDRATE_BACKUP_FAILURE,
  error: CustomError,
}

export type BackupStoreAction =
  | BackupStartAction
  | GenerateBackupFileAction
  | GenerateBackupFileSuccessAction
  | BackupWalletFailAction
  | GenerateRecoveryPhraseAction
  | GenerateRecoveryPhraseLoadingAction
  | GenerateRecoveryPhraseSuccessAction
  | ExportBackupAction
  | ExportBackupSuccessAction
  | BackupCompleteAction
  | HydrateBackupAction
  | HydrateBackupFailAction
