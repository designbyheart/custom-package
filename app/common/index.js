// @flow
export * from './api-constants'
export * from './button-constants'
export * from './message-constants'
export * from './route-constants'
export * from './secure-storage-constants'
export function noop() {}
