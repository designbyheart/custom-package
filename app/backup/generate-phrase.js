// @flow

import React, { PureComponent } from 'react'
import { Image, Dimensions } from 'react-native'
import { StackNavigator } from 'react-navigation'
import {
  Container,
  CustomView,
  CustomText,
  Icon,
  CustomButton,
} from '../components'
import { SHORT_DEVICE } from '../common/styles'
import { genRecoveryPhraseRoute, verifyRecoveryPhraseRoute } from '../common'
import { color } from '../common/styles/constant'
import type {
  GenerateRecoveryPhraseProps,
  GenerateRecoveryPhraseState,
} from './type-backup'
import {
  RECOVERY_PHRASE_CLOSE_TEST_ID,
  SUBMIT_RECOVERY_PHRASE_TEST_ID,
  SUBMIT_RECOVERY_PHRASE_BUTTON_TITLE,
} from './backup-constants'
import styles from './styles'

const { height } = Dimensions.get('window')
const closeImage = require('../images/iconClose.png')
const transparentBands = require('../images/transparentBands.png')
const textBubble = require('../images/textBubble.png')

export class GenerateRecoveryPhrase extends PureComponent<
  GenerateRecoveryPhraseProps,
  GenerateRecoveryPhraseState
> {
  state = {
    recoveryPassphrase: 'one two three',
  }

  verifyRecoveryPhrase = () => {
    const { navigation: { navigate, state } } = this.props
    navigate(verifyRecoveryPhraseRoute, {
      recoveryPassphrase: this.state.recoveryPassphrase,
      initialRoute: state.params.initialRoute,
    })
  }

  static navigationOptions = ({ navigation }) => ({
    headerLeft: <CustomView />,
    headerTitle: <CustomView />,
    headerRight: (
      <CustomView style={[styles.genRecoveryHeader]}>
        <Icon
          medium
          onPress={() => navigation.goBack(null)}
          testID={RECOVERY_PHRASE_CLOSE_TEST_ID}
          iconStyle={[styles.headerCloseIcon]}
          src={closeImage}
        />
      </CustomView>
    ),
    headerStyle: {
      backgroundColor: color.bg.eleventh.color,
      shadowOpacity: 0,
      shadowOffset: {
        height: 0,
      },
      shadowRadius: 0,
      elevation: 0,
      borderBottomWidth: 0,
      shadowColor: 'transparent',
    },
    gesturesEnabled: false,
  })

  render() {
    const { navigation: { state } } = this.props
    return (
      <Container style={[styles.genRecovery]} safeArea>
        <Image source={transparentBands} style={[styles.backgroundImage]} />
        <Container>
          <CustomView>
            <CustomText
              transparentBg
              numberOfLines={1}
              style={[styles.genRecoveryText]}
            >
              Recovery Phrase Generated
            </CustomText>
          </CustomView>
          <CustomView
            center
            style={[styles.genRecoveryWrapper]}
            doubleVerticalSpace={height > SHORT_DEVICE ? true : false}
          >
            <CustomText
              center
              transparentBg
              style={[styles.genRecoveryMessage]}
            >
              This is your recovery phrase. Write it down, and don't share it
              with anyone.
            </CustomText>
          </CustomView>
          <CustomView
            center
            verticalSpace={height > SHORT_DEVICE ? true : false}
          >
            <Image source={textBubble} style={[styles.imageIcon]} />
            <CustomView style={[styles.genRecoveryPhraseContainer]}>
              <CustomText transparentBg style={[styles.genRecoveryPhrase]}>
                {this.state.recoveryPassphrase}
              </CustomText>
            </CustomView>
          </CustomView>
          <CustomView
            center
            verticalSpace={height > SHORT_DEVICE ? true : false}
          >
            <CustomText
              center
              transparentBg
              style={[styles.genRecoverySecondMessage]}
            >
              You will need this recovery phrase to restore from a backup. Keep
              it safe.
            </CustomText>
          </CustomView>
        </Container>
        <CustomView>
          <CustomView center>
            <CustomText
              transparentBg
              center
              style={[styles.genRecoverySmallMessage]}
            >
              Are you sure you wrote it down?
            </CustomText>
          </CustomView>
          <CustomButton
            large={height > SHORT_DEVICE ? true : false}
            onPress={this.verifyRecoveryPhrase}
            testID={SUBMIT_RECOVERY_PHRASE_TEST_ID}
            style={[styles.submitButton]}
            customColor={{
              color: color.bg.eleventh.color,
              fontWeight: '600',
              fontSize: 18,
            }}
            title={SUBMIT_RECOVERY_PHRASE_BUTTON_TITLE}
          />
        </CustomView>
      </Container>
    )
  }
}

export default StackNavigator({
  [genRecoveryPhraseRoute]: {
    screen: GenerateRecoveryPhrase,
  },
})
