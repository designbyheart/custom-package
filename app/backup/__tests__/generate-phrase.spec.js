// @flow

import React from 'react'
import renderer from 'react-test-renderer'
import { GenerateRecoveryPhrase } from '../generate-phrase'
import { getNavigation } from '../../../__mocks__/static-data'
import { settingsRoute } from '../../common'
import { BACKUP_STORE_STATUS } from '../type-backup'

describe('<GenerateRecoveryPhrase />', () => {
  const recoveryPassphrase = 'hello some passphrase'
  const navigation = {
    navigate: jest.fn(),
    goBack: jest.fn(),
    state: {
      params: {
        recoveryPassphrase,
        initialRoute: settingsRoute,
      },
    },
  }

  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <GenerateRecoveryPhrase
          recoveryPassphrase={{
            phrase: recoveryPassphrase,
            salt: 'salt',
            hash: 'hash',
          }}
          generateRecoveryPhrase={jest.fn()}
          recoveryStatus={BACKUP_STORE_STATUS.GENERATE_PHRASE_SUCCESS}
          navigation={navigation}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should match snapshot for loading state', () => {
    const tree = renderer
      .create(
        <GenerateRecoveryPhrase
          recoveryPassphrase={{
            phrase: recoveryPassphrase,
            salt: 'salt',
            hash: 'hash',
          }}
          generateRecoveryPhrase={jest.fn()}
          recoveryStatus={BACKUP_STORE_STATUS.GENERATE_PHRASE_LOADING}
          navigation={navigation}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should match snapshot for error state', () => {
    const tree = renderer
      .create(
        <GenerateRecoveryPhrase
          recoveryPassphrase={{
            phrase: recoveryPassphrase,
            salt: 'salt',
            hash: 'hash',
          }}
          generateRecoveryPhrase={jest.fn()}
          recoveryStatus={BACKUP_STORE_STATUS.GENERATE_PHRASE_FAILURE}
          navigation={navigation}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
