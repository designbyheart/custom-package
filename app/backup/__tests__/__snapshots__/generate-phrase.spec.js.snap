// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenerateRecoveryPhrase /> should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  safeArea={true}
  style={
    Object {
      "backgroundColor": "#4A90E2",
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "flex": 1,
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={Object {}}
    >
      <Text
        allowFontScaling={false}
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            null,
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "alignSelf": "center",
              "fontSize": 20,
              "fontWeight": "600",
              "marginBottom": "2%",
            },
          ]
        }
      >
        Recovery Phrase generated
      </Text>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "alignSelf": "center",
              "fontSize": 16,
              "fontWeight": "500",
              "lineHeight": 20,
              "marginBottom": 5,
              "paddingHorizontal": 20,
            },
          ]
        }
      >
        This is your Recovery Phrase. Write it down, and don't share it with anyone.
      </Text>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Image
        onLayout={[Function]}
        source={1}
        style={
          Array [
            Object {
              "marginBottom": 0,
              "resizeMode": "stretch",
              "width": "100%",
            },
          ]
        }
      />
      <View
        style={
          Object {
            "left": 20,
            "paddingHorizontal": "7%",
            "position": "absolute",
            "top": 20,
            "width": 320,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#333333",
                "color": "#ffffff",
              },
              null,
              null,
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              Object {
                "color": "#4A90E2",
                "flexWrap": "wrap",
                "fontSize": 18,
                "fontWeight": "900",
                "lineHeight": 28,
                "textAlign": "center",
              },
            ]
          }
        >
          hello some passphrase
        </Text>
      </View>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "alignSelf": "center",
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 20,
              "paddingHorizontal": 20,
            },
          ]
        }
      >
        You will need this Recovery Phrase to restore from a backup. Keep it safe.
      </Text>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "fontSize": 14,
              "fontWeight": "bold",
              "lineHeight": 17,
              "marginBottom": "3%",
            },
          ]
        }
      >
        Are you sure you wrote it down?
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          Object {
            "marginLeft": 0,
            "marginRight": 0,
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityLabel="submit-recovery-passphrase"
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 5,
              "elevation": 7,
              "height": 43,
              "marginBottom": 15,
              "marginLeft": "2%",
              "marginRight": "2%",
              "shadowColor": "#777777",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 5,
            },
          ]
        }
        testID="submit-recovery-passphrase"
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#ffffff",
              },
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              Array [
                null,
                Object {
                  "borderRadius": 5,
                  "elevation": 7,
                  "height": 43,
                  "marginBottom": 15,
                  "marginLeft": "2%",
                  "marginRight": "2%",
                  "shadowColor": "#777777",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 5,
                },
              ],
              undefined,
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#4A90E2",
                  },
                  Object {
                    "fontSize": 14,
                  },
                  Object {
                    "fontSize": 18,
                  },
                  undefined,
                  Object {
                    "fontWeight": "600",
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Done
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`<GenerateRecoveryPhrase /> should match snapshot for error state 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  safeArea={true}
  style={
    Object {
      "backgroundColor": "#4A90E2",
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "flex": 1,
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={Object {}}
    >
      <Text
        allowFontScaling={false}
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            null,
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "alignSelf": "center",
              "fontSize": 20,
              "fontWeight": "600",
              "marginBottom": "2%",
            },
          ]
        }
      >
        Recovery Phrase generated
      </Text>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "alignSelf": "center",
              "fontSize": 16,
              "fontWeight": "500",
              "lineHeight": 20,
              "marginBottom": 5,
              "paddingHorizontal": 20,
            },
          ]
        }
      >
        This is your Recovery Phrase. Write it down, and don't share it with anyone.
      </Text>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Image
        onLayout={[Function]}
        source={1}
        style={
          Array [
            Object {
              "marginBottom": 0,
              "resizeMode": "stretch",
              "width": "100%",
            },
          ]
        }
      />
      <View
        style={
          Object {
            "left": 20,
            "paddingHorizontal": "7%",
            "position": "absolute",
            "top": 20,
            "width": 320,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#333333",
                "color": "#ffffff",
              },
              null,
              Object {
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              Object {
                "color": "#4A4A4A",
              },
            ]
          }
        >
          Uh oh... Looks like something went wrong.
        </Text>
      </View>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "alignSelf": "center",
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 20,
              "paddingHorizontal": 20,
            },
          ]
        }
      >
        You will need this Recovery Phrase to restore from a backup. Keep it safe.
      </Text>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "fontSize": 14,
              "fontWeight": "bold",
              "lineHeight": 17,
              "marginBottom": "3%",
            },
          ]
        }
      >
        Are you sure you wrote it down?
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          Object {
            "marginLeft": 0,
            "marginRight": 0,
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityLabel="submit-recovery-passphrase"
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 5,
              "elevation": 7,
              "height": 43,
              "marginBottom": 15,
              "marginLeft": "2%",
              "marginRight": "2%",
              "shadowColor": "#777777",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 5,
            },
          ]
        }
        testID="submit-recovery-passphrase"
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#ffffff",
              },
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              Array [
                null,
                Object {
                  "borderRadius": 5,
                  "elevation": 7,
                  "height": 43,
                  "marginBottom": 15,
                  "marginLeft": "2%",
                  "marginRight": "2%",
                  "shadowColor": "#777777",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 5,
                },
              ],
              undefined,
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#4A90E2",
                  },
                  Object {
                    "fontSize": 14,
                  },
                  Object {
                    "fontSize": 18,
                  },
                  undefined,
                  Object {
                    "fontWeight": "600",
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Done
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`<GenerateRecoveryPhrase /> should match snapshot for loading state 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  safeArea={true}
  style={
    Object {
      "backgroundColor": "#4A90E2",
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "flex": 1,
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={Object {}}
    >
      <Text
        allowFontScaling={false}
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            null,
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "alignSelf": "center",
              "fontSize": 20,
              "fontWeight": "600",
              "marginBottom": "2%",
            },
          ]
        }
      >
        Recovery Phrase generated
      </Text>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "alignSelf": "center",
              "fontSize": 16,
              "fontWeight": "500",
              "lineHeight": 20,
              "marginBottom": 5,
              "paddingHorizontal": 20,
            },
          ]
        }
      >
        This is your Recovery Phrase. Write it down, and don't share it with anyone.
      </Text>
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Image
        onLayout={[Function]}
        source={1}
        style={
          Array [
            Object {
              "marginBottom": 0,
              "resizeMode": "stretch",
              "width": "100%",
            },
          ]
        }
      />
      <View
        style={
          Object {
            "padding": 40,
            "position": "absolute",
            "top": 0,
          }
        }
      />
    </View>
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "alignSelf": "center",
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 20,
              "paddingHorizontal": 20,
            },
          ]
        }
      >
        You will need this Recovery Phrase to restore from a backup. Keep it safe.
      </Text>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            null,
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "fontSize": 14,
              "fontWeight": "bold",
              "lineHeight": 17,
              "marginBottom": "3%",
            },
          ]
        }
      >
        Are you sure you wrote it down?
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          Object {
            "marginLeft": 0,
            "marginRight": 0,
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityLabel="submit-recovery-passphrase"
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 5,
              "elevation": 7,
              "height": 43,
              "marginBottom": 15,
              "marginLeft": "2%",
              "marginRight": "2%",
              "shadowColor": "#777777",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 5,
            },
          ]
        }
        testID="submit-recovery-passphrase"
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#ffffff",
              },
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              Array [
                null,
                Object {
                  "borderRadius": 5,
                  "elevation": 7,
                  "height": 43,
                  "marginBottom": 15,
                  "marginLeft": "2%",
                  "marginRight": "2%",
                  "shadowColor": "#777777",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 5,
                },
              ],
              undefined,
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#4A90E2",
                  },
                  Object {
                    "fontSize": 14,
                  },
                  Object {
                    "fontSize": 18,
                  },
                  undefined,
                  Object {
                    "fontWeight": "600",
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Done
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
